src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LanguageClient/Cohost/HtmlDocumentSynchronizer.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+using Microsoft.CodeAnalysis.Razor.Logging;
+using Microsoft.VisualStudio.LanguageServer.ContainedLanguage;
+
+namespace Microsoft.VisualStudio.Razor.LanguageClient.Cohost;
+
+[Export(typeof(IHtmlDocumentSynchronizer))]
+[method: ImportingConstructor]
+internal sealed partial class HtmlDocumentSynchronizer(
+    LSPDocumentManager documentManager,
+    IHtmlDocumentPublisher htmlDocumentPublisher,
+    ILoggerFactory loggerFactory)
+    : IHtmlDocumentSynchronizer
+{
+    private static readonly Task<bool> s_falseTask = Task.FromResult(false);
```

Comment:
Should there be a `SpecializedTasks` helper class like Roslyn's in Shared.Utilities?

https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/SpecializedTasks.cs,25dbc38fd8ac0df3

Reply:
To be honest, I'm not sure why I even did this. I don't even expect the scenario where its used to be super common that it would show up on a trace. Having said that, creating finding all of the places a SpecializedTasks helper could be used sounds like a good thursday evening job :)